src/Microsoft.DotNet.Darc/DarcLib/Helpers/DependencyFileManager.cs

```diff
@@ -57,7 +60,17 @@ public class DependencyFileManager : IDependencyFileManager
         IVersionDetailsParser versionDetailsParser,
         ILogger logger)
     {
-        _gitClient = gitClient;
+        _gitClientFactory = _ => gitClient;
```

Comment:
Do we need to keep both constructors (one with IGitRepoFactory and the other one with IGitRepo)? I understand this might be needed for seamless transition form one to the other, but having both makes impression we have some misalignments in the actual gitClient lifecycle management as once it's managed through the facotry and once it comes from outside.

Reply:
This would require a follow-up for non-VMR code to use the factory properly and kill the other factories around the code base. I will log an issue but this is better for now, I'd say.

Reply:
I will need to merge this soon to get fixes for Michael so I decided to only do the first half of merging the libraries.